docs: logging: add multidomain sample documentation (#108361)
zephyrproject-rtos/zephyr: samples/subsys/logging/multidomain/README.rst

.. zephyr:code-sample:: logging_multidomain
   :name: Logging in multi-domain environment
   :relevant-api: log_ctrl ipc

Use the multidomain logging feature to aggregate log messages from multiple cores into a single console output via the IPC service.

> ###################################################################################################################################

Reviewer: ```suggestion
   Use the multidomain logging feature to aggregate log messages from multiple
   cores into a single console output via the IPC service.
```

Overview
********
This sample demonstrates the multidomain logging feature in Zephyr on a dual-core SoC. It uses a HOST/REMOTE architecture where the HOST core aggregates log messages from the REMOTE core via the IPC service, so output from both cores appears in a single terminal.

Requirements
************
This sample requires a board with dual-core support and IPC capability.

> The following targets are supported:

Reviewer: ```suggestion
The following targets are supported:

```

* ``nrf5340dk/nrf5340/cpuapp``
* ``nrf5340bsim/nrf5340/cpuapp``

Building and Running
********************
This sample can be built and flashed on the nrf5340dk as follows:

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/logging/multidomain
   :board: nrf5340dk/nrf5340/cpuapp
   :goals: build flash
   :west-args: --sysbuild
   :compact:

Sample Output
=============

The following lines are expected to appear in the HOST UART output
(the order may vary):

.. code-block:: console

   Hello World! nrf5340dk
   <inf> app: IPC-service HOST [INST 1] demo started
   <inf> app: loop: 0

>    <inf> app: ipc open

Reviewer: ```suggestion
   <inf> app: ipc open 0
```

.. zephyr:code-sample:: logging_multidomain
   :name: Logging in multi-domain environment
   :relevant-api: log_ctrl ipc

 Use the multidomain logging feature to aggregate log messages from multiple

>  cores into a single console output via the IPC service.

Reviewer: ```suggestion
   Use the multidomain logging feature to aggregate log messages from multiple
   cores into a single console output via the IPC service.
```
The indent needs to be 3 spaces

.. zephyr:code-sample:: logging_multidomain
   :name: Logging in multi-domain environment
   :relevant-api: log_ctrl ipc

> Logging in multi-domain environment

Reviewer: ```suggestion
Use the multidomain logging feature to aggregate log messages from multiple cores into a single console output via the IPC service.
```

###################################

> This code sample shows how to use the multidomain logging feature in Zephyr.

Reviewer: ```suggestion

```

Overview
********
This sample utilizes a HOST/REMOTE architecture where the primary core (HOST) aggregates and processes log messages generated by the secondary cores (REMOTE) via the IPC service.This sample utilizes a HOST/REMOTE architecture where the primary core (HOST) aggregates and processes log messages generated by the secondary cores (REMOTE) via the IPC service.

> It utilizes the Inter-Processor Communication (IPC) service to route log messages between domains, allowing you to collect and view logs from multiple cores in a single terminal output.

Reviewer: ```suggestion
This sample demonstrates the multidomain logging feature in Zephyr on a dual-core SoC. It uses a HOST/REMOTE architecture where the HOST core aggregates log messages from the REMOTE core via the IPC service, so output from both cores appears in a single terminal.
```

Requirements
************
The sample requires a board with multiple cores and IPC support (such as the nRF5340 DK).It relies on the following key configuration options being enabled:
* ``CONFIG_LOG_MULTIDOMAIN``
* ``CONFIG_IPC_SERVICE``

> * ``CONFIG_LOG_LINK_IPC_SERVICE``

Reviewer: ```suggestion
This sample requires a board with dual-core support and IPC capability.
The following targets are supported:

* ``nrf5340dk/nrf5340/cpuapp``
* ``nrf5340bsim/nrf5340/cpuapp``
```

Building and Running
********************
This sample can be found under ``samples/subsys/logging/multidomain`` in the Zephyr source tree.

You can build and run this sample on the nRF5340 DK using the following commands:

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/logging/multidomain
   :board: nrf5340dk/nrf5340/cpuapp
   :compact:
   :goals: build flash

>    :west-args: --sysbuild

Reviewer: ```suggestion
This sample can be built and flashed on the nrf5340dk as follows:

.. zephyr-app-commands::
   :zephyr-app: samples/subsys/logging/multidomain
   :board: nrf5340dk/nrf5340/cpuapp
   :goals: build flash
   :west-args: --sysbuild
   :compact:
```

Sample Output
*************

When the sample is run, the test harness expects the following sequence in the console output:

.. code-block:: console

   Hello World! nrf5340dk
   app: IPC-service HOST [INST 1] demo started
   app: loop: 0
   app: ipc open
   app: wait for bound
   app: bounded
   app: REMOTE [1]: 0
   app: HOST [1]: 1

>    app: IPC-service HOST [INST 1] demo ended.

Reviewer: ```suggestion
=============

The following lines are expected to appear in the HOST UART output
(the order may vary):

.. code-block:: console

   Hello World! nrf5340dk
   <inf> app: IPC-service HOST [INST 1] demo started
   <inf> app: loop: 0
   <inf> app: ipc open 0
   <inf> app: wait for bound
   <inf> app: ipc open 0
   <inf> app: bounded
   <inf> app: REMOTE [1]: 0
   <inf> app: HOST [1]: 1
   ...
   <inf> app: IPC-service HOST [INST 1] demo ended.
```
